src/main/kotlin/com/hibob/kotlinBasic/StoreService.kt

```diff
@@ -0,0 +1,47 @@
+package com.hibob.kotlinBasic
+
+import org.springframework.stereotype.Component
+import java.time.LocalDate
+import java.time.LocalDateTime
+
+@Component
+class StoreService {
+
+    fun pay(cart: List<Cart>, payment: Payment): Map<String, Check> {
+        return cart.map { cart -> cart.clientId to checkout(cart, payment) }.toMap()
```

Comment:
1. you call the same name for local var (inside the map) as the List<Cart> name.
2. you can use associate function that return map:
```
    fun pay(cart: List<Cart>, payment: Payment): Map<String, Check> {
        return cart.associate { it.clientId to checkout(it, payment) }
    }
```